ui/src/features/stage/edit-stage-modal.tsx

```diff
@@ -29,24 +28,31 @@ const formSchema = z.object({
 });
 
 export const EditStageModal = ({ visible, hide, projectName, stageName }: Props) => {
-  const { data } = useQuery(listStages, { project: projectName });
-  const stage = data?.stages.find((item) => item.metadata?.name === stageName);
+  const { data, isLoading } = useQuery(getStage, {
+    project: projectName,
+    name: stageName,
+    format: RawFormat.YAML
+  });
+
+  const manifest = new TextDecoder().decode(
```

Comment:
This is the second time this logic is used - maybe worth creating a utility function.